indra/newview/app_settings/shaders/class1/environment/waterFogF.glsl

```diff
@@ -108,3 +110,35 @@ vec4 applyWaterFogViewLinear(vec3 pos, vec4 color)
     return applyWaterFogViewLinearNoClip(pos, color);
 }
 
+// for post deferred shaders, apply sky and water fog in a way that is consistent with
+// the deferred rendering haze post effects
+vec4 applySkyAndWaterFog(vec3 pos, vec3 additive, vec3 atten, vec4 color)
+{
+    bool eye_above_water = dot(vec3(0), waterPlane.xyz) + waterPlane.w > 0.0;
+    bool obj_above_water = dot(pos.xyz, waterPlane.xyz) + waterPlane.w > 0.0;
```

Comment:
eye_above_water: Dot product of 0 is 0. Could remove this.
More generally, I'm not sure what's going on in `LLSettingsVOWater::applySpecial`, where waterPlane is defined. The naming convention for `ep` and `enorm` seem inconsistent. `ep` is in eye space, but `enorm` is not. So, I wonder what is the reasoning behind `-ep.dot(enorm)`, if `ep` and `enorm` are not in the same coordinate system?